### ISSUE TITLE
[Accessibility]: Homepage search suggestions should use semantic markup

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43141]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34189

### Problem description

It is currently a bit difficult for screen reader users to understand the structure of the new homepage search suggestions as the groupings are not clear, with no use of headings or lists.

### Expected behavior

The homepage search suggestions use headings and lists to group items more semantically. Example:

![CleanShot 2022-10-18 at 15 40 32@2x](https://user-images.githubusercontent.com/206864/196558656-24154ab3-8251-4062-b5eb-d618666f7839.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard
